File: Java/1829_Maximum_XOR_for_Each_Query.java
class Solution {
    public int[] getMaximumXor(int[] nums, int maximumBit) {
        int n = nums.length;
        int[] result = new int[n];
        int m = (int)Math.pow(2, maximumBit) - 1;
        for(int i = 0; i < n; i++){
            if(i == 0){
                result[n-1-i] = nums[i] ^= m;
            }
            else{
                result[n-1-i] = nums[i] ^= nums[i-1];
            }
        }
        return result;
    }
}

File: CS/1829_Maximum_XOR_for_Each_Query.cs
public class Solution {
    public int[] GetMaximumXor(int[] nums, int maximumBit) {
        int n = nums.Length;
        int[] result = new int[n];
        //int x = (int) Math.Pow(2, maximumBit) - 1;
        int x = (1 << maximumBit) - 1;
        int temp = x;
        for(int i = 0; i < n; i++){
            temp ^= nums[i];
            result[n-1-i] = temp;
        }
        return result;
    }
}
